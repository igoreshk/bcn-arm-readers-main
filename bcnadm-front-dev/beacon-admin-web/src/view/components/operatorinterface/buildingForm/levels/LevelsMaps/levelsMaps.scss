/* stylelint-disable declaration-no-important */
.levelsMaps {
  .levelsDropzones {
    margin-top: 20px;

    .scrollbar {
      height: 300px !important;
    }

    .scrollbarContent {
      padding: 0 10px;
    }

    .createLevelContainer {
      display: flex;
      align-items: center;
      margin-bottom: 15px;
    }

    .levelNumber {
      min-width: 40px;
      margin-right: 30px;

      @media screen and (max-width: 500px) {
        margin-right: 10px;
      }
    }

    .dropzoneContainer {
      margin-right: 30px;
      flex-grow: 1;

      @media screen and (max-width: 500px) {
        margin-right: 10px;
      }
    }

    .dropzone {
      height: 60px;
      border: dashed 1px;
      border-color: #39C2D7;
      cursor: pointer;
      display: flex;
      justify-content: center;
      align-items: center;

      @media screen and (max-width: 700px) {
        height: 150px;
      }
    }

    .dropzoneText {
      color: #39C2D7;
      text-align: center;
      max-width: 260px;
      overflow: hidden;
      text-overflow: ellipsis;

      @media screen and (max-width: 700px) {
        display: none;
      }

      @media screen and (max-width: 600px) {
        font-size: 14px;
      }

      @media screen and (max-width: 425px) {
        font-size: 12px;
      }
    }

    .dropzoneContent {
      width: 100%;
      height: inherit;
      display: flex;
      align-items: center;
      justify-content: center;
    }

    .preview {
      width: 80px;
      height: inherit;
      object-fit: cover;
      margin-left: 30px;

      @media screen and (max-width: 700px) {
        width: inherit;
        margin-left: 0;
      }
    }

    .preview:hover {
      @media screen and (min-width: 701px) {
        position: fixed;
        height: 60%;
        width: 60%;
        top: 20%;
        left: 20%;
        z-index: 1;
        margin: 0;
      }
    }

    .error {
      color: #FF0000;
      font-size: 12px;
    }

    @media screen and (max-width: 500px) {
      .removeButton {
        padding: 0;
      }
    }

    .addLevelButtonContainer {
      display: flex;
      justify-content: center;
      margin-top: 10px;
    }
  }

  .buttonsContainer {
    display: flex;
    justify-content: flex-end;
    margin-top: 20px;

    .submitButton {
      margin-left: 10px;
    }
  }
}