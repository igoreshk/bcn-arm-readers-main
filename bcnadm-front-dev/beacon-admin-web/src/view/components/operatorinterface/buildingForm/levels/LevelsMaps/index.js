import { ERROR, SUCCESSFUL } from 'src/utils/popUpConsts';
import React, { Component } from 'react';
import { Button } from '@material-ui/core';
import { FaBuilding as BuildingIcon } from 'react-icons/fa';
import PropTypes from 'prop-types';

import Logo from '../../styles/Logo';
import LevelsDropzones from '../LevelsDropzone';

import './levelsMaps.scss';

export default class LevelsMaps extends Component {
  state = {
    isSubmitInProgress: false
  };

  changeStatusOfSubmit = (value) => {
    this.setState({
      isSubmitInProgress: value
    });
  };

  handleOnSubmit = async (values) => {
    const { handleSubmit, addPopupMessageText, translate, closeDialog } = this.props;

    try {
      this.setState({
        isSubmitInProgress: true
      });
      const result = await handleSubmit(values);
      if (result.building && result.levels) {
        addPopupMessageText(translate('building.messages.success'), SUCCESSFUL);
        closeDialog();
      } else {
        throw Error(ERROR);
      }
    } catch (error) {
      this.setState(
        {
          isSubmitInProgress: false
        },
        () => addPopupMessageText(translate('building.messages.serverError'), error.message)
      );
    }
  };

  render() {
    const {
      state: { isSubmitInProgress },
      props: { allowedToProceed, back, levels, translate, isNew, updateLevels, removeLevelField },
      handleOnSubmit
    } = this;

    return (
      <form onSubmit={handleOnSubmit} className="levelsMaps">
        <Logo
          title={translate(isNew ? 'building.create.titleLevels' : 'building.update.titleLevels')}
          icon={<BuildingIcon className="logo" />}
        />
        <LevelsDropzones
          changeStatusOfSubmit={this.changeStatusOfSubmit}
          updateLevels={updateLevels}
          removeLevelField={removeLevelField}
          levels={levels}
          translate={translate}
        />
        <div className="buttonsContainer">
          <Button onClick={back}>{translate('building.back')}</Button>
          <Button
            disabled={!allowedToProceed || isSubmitInProgress}
            type="submit"
            className="submitButton"
            variant="contained"
            color="primary"
          >
            {translate(isNew ? 'building.create.save' : 'building.update.save')}
          </Button>
        </div>
      </form>
    );
  }
}

LevelsMaps.propTypes = {
  allowedToProceed: PropTypes.bool.isRequired,
  back: PropTypes.func.isRequired,
  closeDialog: PropTypes.func.isRequired,
  handleSubmit: PropTypes.func.isRequired,
  isNew: PropTypes.bool.isRequired,
  levels: PropTypes.arrayOf(PropTypes.object),
  removeLevelField: PropTypes.func.isRequired,
  translate: PropTypes.func.isRequired,
  updateLevels: PropTypes.func.isRequired
};
