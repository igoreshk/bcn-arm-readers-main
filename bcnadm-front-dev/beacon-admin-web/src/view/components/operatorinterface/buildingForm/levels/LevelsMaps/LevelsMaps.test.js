import LevelsMaps from 'src/view/components/operatorinterface/buildingForm/levels/LevelsMaps/index';
import React from 'react';
import { shallow } from 'enzyme';

const props = {
  back: jest.fn(),
  closeDialog: jest.fn(),
  translate: jest.fn().mockReturnValue(''),
  allowedToProceed: true,
  handleSubmit: jest.fn(),
  isNew: true,
  levels: [],
  removeLevelField: jest.fn(),
  updateLevels: jest.fn()
};

const wrapper = shallow(<LevelsMaps {...props} />);

describe('LevelsMaps test', () => {
  it('check props matches', () => expect(wrapper.length).toEqual(1));
  it('snapshot for LevelsMaps', () => expect(wrapper).toMatchSnapshot());
  it('check if handleSubmit was called', () => {
    const handleSubmit = jest.spyOn(wrapper.instance().props, 'handleSubmit');
    wrapper.instance().handleOnSubmit();
    expect(handleSubmit).toHaveBeenCalled();
  });

  it('State isSubmitInProgress is false when mount component', () => {
    expect(wrapper.state().isSubmitInProgress).toBeFalsy();
  });

  it('State isSubmitInProgress changes when call changeStatusOfSubmit', () => {
    wrapper.instance().changeStatusOfSubmit(true);
    expect(wrapper.state().isSubmitInProgress).toBeTruthy();
  });
});
