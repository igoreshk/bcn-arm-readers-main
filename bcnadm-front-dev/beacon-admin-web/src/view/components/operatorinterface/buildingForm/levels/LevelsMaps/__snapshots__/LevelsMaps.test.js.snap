// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LevelsMaps test snapshot for LevelsMaps 1`] = `
<form
  className="levelsMaps"
  onSubmit={[Function]}
>
  <Logo
    icon={
      <FaBuilding
        className="logo"
      />
    }
    title=""
  />
  <LevelsDropzones
    changeStatusOfSubmit={[Function]}
    levels={Array []}
    removeLevelField={[MockFunction]}
    translate={
      [MockFunction] {
        "calls": Array [
          Array [
            "building.create.titleLevels",
          ],
          Array [
            "building.back",
          ],
          Array [
            "building.create.save",
          ],
        ],
        "results": Array [
          Object {
            "type": "return",
            "value": "",
          },
          Object {
            "type": "return",
            "value": "",
          },
          Object {
            "type": "return",
            "value": "",
          },
        ],
      }
    }
    updateLevels={[MockFunction]}
  />
  <div
    className="buttonsContainer"
  >
    <WithStyles(ForwardRef(Button))
      onClick={[MockFunction]}
    />
    <WithStyles(ForwardRef(Button))
      className="submitButton"
      color="primary"
      disabled={false}
      type="submit"
      variant="contained"
    />
  </div>
</form>
`;
